### PR TITLE
[FEATURE] Add a github ci

### PR DESCRIPTION
**Pull request by Lord-Giganticus**
This is part of the changes/requests made in #18 before I accidentally messed it up 🤣 

Changes:
* Added a github ci file (dotnet.yml) to build the C# program on all major operation systems. (x64 bit only, due to ms being weird)
* Closes #18

### CODE REVIEW

**Comment by RoadrunnerWMC** [comment]
Hey, sorry for the super late response... Just a few questions/comments about the config file:

- Why do you have `include-prerelease: true` for .NET setup?
- I generally prefer `on: [push]` for the action trigger, since I find it useful to see when other branches are failing to build before they get merged into master :P
- Could you add a name string to the shell-command step? Just so it'd look a bit nicer in the web UI.
- On that note, "`.NET`" isn't a very descriptive name for the whole action -- maybe something like "`Build`" or "`Build with .NET`"?

Thanks!

**Comment by Lord-Giganticus** [comment]
> Hey, sorry for the super late response... Just a few questions/comments about the config file:
> 
> * Why do you have `include-prerelease: true` for .NET setup?
> * I generally prefer `on: [push]` for the action trigger, since I find it useful to see when other branches are failing to build before they get merged into master :P
> * Could you add a name string to the shell-command step? Just so it'd look a bit nicer in the web UI.
> * On that note, "`.NET`" isn't a very descriptive name for the whole action -- maybe something like "`Build`" or "`Build with .NET`"?
> 
> Thanks!

Hello again.
Here's a explanation of some things.
> * Why do you have `include-prerelease: true` for .NET setup?
As far as I know dotnet 6.0 is not yet supported on without that line, until they do it has to stay.
> * I generally prefer `on: [push]` for the action trigger, since I find it useful to see when other branches are failing to build before they get merged into master :P
> * On that note, "`.NET`" isn't a very descriptive name for the whole action -- maybe something like "`Build`" or "`Build with .NET`"?
Blame github's yaml generation on that,
> * Could you add a name string to the shell-command step? Just so it'd look a bit nicer in the web UI.
> * On that note, "`.NET`" isn't a very descriptive name for the whole action -- maybe something like "`Build`" or "`Build with .NET`"?
Sure. I'll make some edits to it tonight or tommorow. 👍🏻

**Comment by RoadrunnerWMC** [comment]
^ Accidentally pushed that test commit to the wrong repo, sorry

**Comment by RoadrunnerWMC** [comment]
> As far as I know dotnet 6.0 is not yet supported on without that line, until they do it has to stay.

[I just tested removing it, and it seems to work fine.](https://github.com/RoadrunnerWMC/Kamek/runs/6299483985) I also can't find any documentation indicating that the flag is currently necessary for dotnet 6, though the documentation for this stuff is admittedly a little hard to follow. Maybe they promoted it out of prerelease status recently.

> Blame github's yaml generation on that.

Sure, but would you still be open to changing it? (I notice you only included the second of those two bullet points in your last block-quote.)

**Comment by Lord-Giganticus** [comment]
> > As far as I know dotnet 6.0 is not yet supported on without that line, until they do it has to stay.
> 
> 
> 
> [I just tested removing it, and it seems to work fine.](https://github.com/RoadrunnerWMC/Kamek/runs/6299483985) I also can't find any documentation indicating that the flag is currently necessary for dotnet 6, though the documentation for this stuff is admittedly a little hard to follow. Maybe they promoted it out of prerelease status recently.
> 
> 
> 
> > Blame github's yaml generation on that.
> 
> 
> 
> Sure, but would you still be open to changing it? (I notice you only included the second of those two bullet points in your last block-quote.)

Oh, when DotNet 6.0 released prerelease was still needed. Good to know it's supported!

I am open to changing the name of the action and will try to make changes tonight.

**Comment by Lord-Giganticus** [comment]
97982d4 has been pushed to implement requested changes

**Comment by Lord-Giganticus** [comment]
97982d4 has been pushed to implement requested changes

**Comment by Lord-Giganticus** [comment]
CI success example [here](https://github.com/Lord-Giganticus/Kamek/actions/runs/2273568832)

**Comment by RoadrunnerWMC** [comment]
Thanks!

I just tested the Linux artifact from the run you linked, and noticed a couple other issues while doing so, though I don't think either of them should block this PR:

The first is one which I've previously run into with other projects: since zip files don't preserve Unix executable flags (+x), Linux and macOS users need to `chmod +x Kamek` before they can run it. I've solved this in the past by having the workflow upload the artifact as .tar.gz rather than a loose executable, but that probably won't work on Windows runners because files have no executable flags even while sitting in the filesystem. So I'm not sure how to solve this in the workflow's current design.

The other issue is that the artifacts contain *only* the built executables and required libraries, and none of the other supporting files from the repo (readme, examples folder, etc). The executables are useful, of course, but it does mean these artifacts aren't really suitable to use as proper full Kamek releases for the Releases page. But on the other hand, creating releases would also require improving the example build scripts (providing equivalent batch and bash scripts for all examples, de-hardcoding the "`..\Kamek\bin\Debug\Kamek`" paths, etc), which is out of scope for this PR. So that can be addressed separately some other time.

Anyway, just wanted to document those ideas while they were on my mind. The PR LGTM -- I'll go ahead and merge it. Thanks!
